### PR TITLE
Magazine Repack - Add ability to disable repacking via class

### PR DESCRIPTION
**Pull request by Mike-MF**
**When merged this pull request will:**
- Adds optional ability to disable magazine repacking by `CfgMagazines` entry.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/magazinerepack/functions/fnc_getMagazineChildren.sqf [hunk 1]
```diff
@@ -25,9 +25,10 @@ private _unitMagCounts = [];
     _x params ["_xClassname", "_xCount", "_xLoaded", "_xType"];
 
     private _xFullMagazineCount = getNumber (configFile >> "CfgMagazines" >> _xClassname >> "count");
+    private _isRepackDisabled = getNumber (configFile >> "CfgMagazines" >> _magazine >> "ace_disableRepacking") == 1;
```
Comment (PabstMirror) [inline addons/magazinerepack/functions/fnc_getMagazineChildren.sqf:28]:
```suggestion
    private _isRepackDisabled = getNumber (configFile >> "CfgMagazines" >> _xClassname >> "ace_disableRepacking") == 1;
```

File: addons/magazinerepack/functions/fnc_canRepackMagazine.sqf [hunk 2]
```diff
@@ -19,11 +19,13 @@
 params ["_unit", "_magazine"];
 
 private _maxAmmoCount = getNumber (configFile >> "CfgMagazines" >> _magazine >> "count");
+private _isRepackDisabled = getNumber (configFile >> "CfgMagazines" >> _magazine >> "ace_disableRepacking") == 1;
```
Comment (BrettMayson) [inline addons/magazinerepack/functions/fnc_canRepackMagazine.sqf:22]: Can check this before the count and just an exitWith, avoid two config lookups when disabled

incredibly minor impact though

**Comment by Mike-MF** [comment]
Someone should look at the documentation, No idea if I did it right or not for a new file.

File: docs/wiki/framework/magazine-repack-framework.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## 1. Config Values
+
+### 1.1 Disable Magazine Repack
+
+```cpp
+class CfgMagazines {
+    class MyMagazine {
+        // Disables the ability to repack this magazine
+        ace_disableRepacking = 1;
```
Comment (johnb432) [inline docs/wiki/framework/magazine-repack-framework.md:23]: Why not `QGVAR(disableRepacking)`?

**Review by johnb432 — COMMENTED** [review]
Do you have a specific use case in mind? Atm I can't really see any, I would be happy to learn about the scenarios where you'd want/need to use this.

File: docs/wiki/framework/magazine-repack-framework.md (hunk 3 above)
Comment (Mike-MF) [reply docs/wiki/framework/magazine-repack-framework.md:23]: Followed the general standard from the other ones I know of

**Comment by Mike-MF** [comment]
> Do you have a specific use case in mind? Atm I can't really see any, I would be happy to learn about the scenarios where you'd want/need to use this.

Any item you track the count of that you wouldn't be able to "repack", for example Brett mentioned that you can repack UAV batteries which .. probably shouldn't work.

Survival mods/scenarios that utilise ACE would also benefit from this.

**Comment by MiszczuZPolski** [comment]
> Do you have a specific use case in mind? Atm I can't really see any, I would be happy to learn about the scenarios where you'd want/need to use this.
I guess you can also use it to disable repacking ACE_Painkillers?

**Comment by Mike-MF** [comment]
you can, but painkillers actually makes sense as it's a pill bottle, just pour one into the other and ignore the bullet clicky sounds.

**Comment by MiszczuZPolski** [comment]
> you can, but painkillers actually makes sense as it's a pill bottle, just pour one into the other and ignore the bullet clicky sounds.

yet model is pill box 🦘 tho i already see use case for this in KAM

**Comment by rautamiekka** [comment]
> repack UAV batteries

And even if you had a small enough portable charger to transfer the energy between the batteries, it'd take far too long to be of any use (something like 1-4 hours (I think) when the battery is like 15-30%), which'd break any resemblance of realism.

> > you can, but painkillers actually makes sense as it's a pill bottle, just pour one into the other and ignore the bullet clicky sounds.
> 
> yet model is pill box 🦘 tho i already see use case for this in KAM

I doubt the pills would be loose in the box, instead in some kinda holders.

**Comment by jonpas** [comment]
> I doubt the pills would be loose in the box, instead in some kinda holders.

They do come loose in the box in the US.

**Review by jonpas — APPROVED** [review]

**Review by PabstMirror — APPROVED** [review]

File: addons/magazinerepack/functions/fnc_canRepackMagazine.sqf [hunk 4]
```diff
@@ -18,6 +18,9 @@
 
 params ["_unit", "_magazine"];
 
+// Exit if repack is disabled for this magazine.
+if (getNumber (configFile >> "CfgMagazines" >> _magazine >> "ace_disableRepacking") == 1) exitWith {};
```
Comment (johnb432) [inline addons/magazinerepack/functions/fnc_canRepackMagazine.sqf:22]:
```suggestion
if (getNumber (configFile >> "CfgMagazines" >> _magazine >> "ace_disableRepacking") == 1) exitWith {false};
```

**Review by johnb432 — CHANGES_REQUESTED** [review]

File: addons/magazinerepack/functions/fnc_getMagazineChildren.sqf [hunk 5]
```diff
@@ -25,9 +25,10 @@ private _unitMagCounts = [];
     _x params ["_xClassname", "_xCount", "_xLoaded", "_xType"];
 
     private _xFullMagazineCount = getNumber (configFile >> "CfgMagazines" >> _xClassname >> "count");
+    private _isRepackDisabled = getNumber (configFile >> "CfgMagazines" >> _xClassname >> "ace_disableRepacking") == 1;
```
Comment (johnb432) [inline addons/magazinerepack/functions/fnc_getMagazineChildren.sqf:28]: Micro-optimisation
```suggestion
    private _configMagazine = getNumber (configFile >> "CfgMagazines" >> _xClassname);
    private _xFullMagazineCount = getNumber (_configMagazine >> "count");
    private _isRepackDisabled = getNumber (_configMagazine >> "ace_disableRepacking") == 1;
```

File: docs/wiki/framework/magazine-repack-framework.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+---
+layout: wiki
+title: Magazine Repack Framework
+description: Explains how to set-up the disable repack.
```
Comment (johnb432) [inline docs/wiki/framework/magazine-repack-framework.md:4]:
```suggestion
description: Explains how to disable repacking.
```

File: docs/wiki/framework/magazine-repack-framework.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+mod: ace
+version:
+  major: 3
+  minor: 18
+  patch: 0
+---
+
+## 1. Config Values
+
+### 1.1 Disable Magazine Repack
```
Comment (johnb432) [inline docs/wiki/framework/magazine-repack-framework.md:17]:
```suggestion
### 1.1 Disable Magazine Repacking
```

**Review by johnb432 — APPROVED** [review]

**Review by BrettMayson — APPROVED** [review]

**Pull request merged**